internal/services/firewall/firewall_resource.go

```diff
@@ -356,6 +395,14 @@ func resourceFirewallCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) e
 		defer locks.UnlockByName(id.FirewallPolicyName, AzureFirewallPolicyResourceName)
 	}
 
+	if autoscaleConfiguration := d.Get("autoscale").(*pluginsdk.Set).List(); len(autoscaleConfiguration) > 0 {
+		configuration, err := expandAutoscaleConfiguration(autoscaleConfiguration)
```

Comment:
Since this is now a TypeList, and normally we will do below and do the checks inside the expand function if needed:

```suggestion
	configuration, err := expandAutoscaleConfiguration(d.Get("autoscale").([]any))
```